src/mailingui/components/buttons/Button.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { FC, ReactNode, CSSProperties } from "react";
+import { Button as ReactEmailButton } from "./ReactEmailButtonFork";
+
+type sizes = "xs" | "sm" | "md" | "lg" | "xl" | "2xl";
+
+interface ButtonProps {
+  href: string;
+  children: ReactNode;
+  backgroundColor?: CSSProperties["color"];
+  color?: CSSProperties["color"];
+  borderColor?: CSSProperties["color"];
+  rounded?: number;
+  size?: sizes;
+}
+
+const Button: FC<ButtonProps> = ({
+  rounded = 8,
+  href,
+  backgroundColor,
+  color,
+  size = "md",
+  borderColor,
+  children,
+}) => {
+  const styles: CSSProperties = {
+    backgroundColor: backgroundColor ?? "#24292e",
+    color: color ?? "#ffffff",
+    borderRadius: rounded,
+    fontSize: sizesEnum[size].fontSize,
+    border: borderColor ? `1px solid ${borderColor}` : "none",
+  };
+
+  return (
+    <ReactEmailButton
+      pX={sizesEnum[size].paddingX}
+      pY={sizesEnum[size].paddingY}
+      href={href}
+      style={styles}
+    >
+      {children}
+    </ReactEmailButton>
+  );
+};
+
+const sizesEnum: Record<
```

Comment:
I would prefer to call this variant rather than enum, and also play around with a structure like this perhaps:

```tsx
// When I consume, I would like something like this
const style = {
  color: variants.color[color]
  fontSize: variants.fontSize[size]
  paddingInline: variants.padding[size].x
  paddingBlock: variants.padding[size].y
  borderRadius: variants.borderRadius[rounded]
}

// I know it is a bit repetitive, but you are accessing the property and then the variant from that property.

// Like variants[<property>][<variant>]

const variants = {
  color: {
    primary: { //... },
    secondary: { //... },
  },
  fontsize: {
    sm: { //... },
    md: { //... },
    lg: { //... },
  },
  borderRadius: {
    sm: { //... },
    md: { //... },
    lg: { //... },
  }
  //..
}
```

Check CVA
https://cva.style/docs/getting-started#your-first-component

Also TailwindCSS follows a similar convention
https://tailwindcss.com/docs/customizing-spacing#overriding-the-default-spacing-scale

We also have to discuss and see how would this fit with having the themes in place.

Reply:
I think we can make these changes in the next PR, where we introduce the ThemeProvider, right?

I'll make a note to not forget to consider this when doing the PR.